R/net.mod.updater.R

```diff
@@ -0,0 +1,146 @@
+#' Update list code{x} using the elements of list code{new.x}
+#'
+#' @param x a list
+#' @param new.x a list
+#'
+#' @return the full code{x} list with the modifications added by code{new.x}
+#'
+#' @details
+#' This function updates list code{x} by name. If code{x} and code{new.x} elements are not
+#' named, the function will not work properly.
+#' If a function is provided to replace an element that was originaly not a
+#' function, this function will be applied to the original value.
+#'
+#' @keywords internal
+update_list <- function(x, new.x) {
+  for (nm in names(new.x)) {
+    if (is.list(new.x[[nm]])) {
+      x[[nm]] <- update_list(x[[nm]], new.x[[nm]])
+    } else if (is.function(new.x[[nm]]) && ! is.function(x[[nm]])) {
+      x[[nm]] <- new.x[[nm]](x[[nm]])
+    } else {
+      x[[nm]] <- new.x[[nm]]
+    }
+  }
+
+  return(x)
+}
+
+#' Module to modify the parameters during the simulation
+#'
+#' @param dat Master list object containing a \code{networkDynamic} object and
+#'        other initialization information passed from \code{\link{netsim}}.
+#' @param at Current time step.
+#'
+#' @return the updated dat Master list object.
+#'
+#' @details
+#' if a list code{param.updater.list} is present in the parameters, this module will
+#' update the code{param} list with new values at given timesteps.
+#' An updater is a list containing an code{at} element governing when the
+#' changes will happen, an optional code{verbose} boolean controlling whether to
+#' output a message when a change is made (default = TRUE) and a code{param}
+#' named list with the names being the same as the paramter names and the new
+#' value to update with.
+#' If the new value is a function but the old one is not, the
+#' function will be applied to the current element (see example) .
+#'
+#' @examples
+#' \dontrun{
+#'
+#' # Create the parame.updater.list
+#' param.updater.list <- list(
+#'   # this is one updater
+#'   list(
+#'     at = 10,
+#'     param = list(
+#'       hiv.test.rate = rep(0.0128, 3),
+#'       trans.scale = c(1.61, 0.836, 0.622)
+#'     )
+#'   ),
+#'   # this is another updater
+#'   list(
+#'     at = 12,
+#'     verbose = TRUE,
+#'     param = list(
+#'       hiv.test.rate = function(x) x * 3,
+#'       trans.scale = function(x) x^2 / 3
+#'     )
+#'   )
+#' )
+#'
+#'  # Do not forget to add it to `param`
+#'  param <- param.net(
+#'    inf.prob = 0.3,
+#'    act.rate = 0.5,
+#'    hiv.test.rate = rep(0.256, 3),
+#'    trans.scale = c(1, 2, 3),
+#'    param.updater.list = param.updater.list
+#'  )
+#'
+#' # Enable the module in `control`
+#'  control <- control.net(
+#'    type = NULL, # must be NULL as we use a custom module
```

Comment:
because of this, the updater module will be the only one that runs. can you use a simple closed-population SI model to demonstrate instead? For example, you need to define an infection module.

Reply:
done